### PR TITLE
Fix for broken GSI documentation

### PR DESCRIPTION
**Pull request by a-atri**
## Jira

* AV-92771

## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

### How was this change tested and do you have evidence? _**(REQUIRED: Select at least 1)**_

- [ ] Manually tested
- [ ] Unit tested
- [ ] Acceptance tested
- [x] Unable to test / will not test (Please provide comments in section below)

### Testing

<details open>
  <summary>Testing</summary>
  
</details>

## Required Checklist:

- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if required)
- [x] I have run make fmt and formatted my code
- [x] I have made sure that no schema field is marked with both requiresReplace and computed
## Further comments

### CODE REVIEW

**Review by SaicharanCB — DISMISSED** [review]

**Review by PaulomeeCb — APPROVED** [review]
lgtm 👍 , would request review from @l0n3star for cross-checking GSI.

File: docs/resources/cluster.md [hunk 1]
```diff
@@ -75,7 +76,7 @@ Required:
 Required:
 
 - `compute` (Attributes) (see [below for nested schema](#nestedatt--service_groups--node--compute))
-- `disk` (Attributes) The 'storage' and 'IOPS' fields are required for AWS. For Azure, only the 'disktype' field is required, and for Ultra, you can provide all three fields. In the case of GCP, only 'pd ssd' disk type is available, and you cannot set the 'IOPS' field. (see [below for nested schema](#nestedatt--service_groups--node--disk))
+- `disk` (Attributes) The 'storage' and 'IOPS' fields are required for AWS. For Azure, only the 'disktype' field is required, and for Ultra disk type, you can provide all 3 - storage, iops and autoexpansion fields. For Premium type, you can only provide the autoexpansion field, others can't be set.In the case of GCP, only 'pd ssd' disk type is available, and you cannot set the 'IOPS' field. (see [below for nested schema](#nestedatt--service_groups--node--disk))
```
Comment (Rebecca-Martinez007) [inline docs/resources/cluster.md:79]:
```suggestion
- `disk` (Attributes) For AWS, the 'storage' and 'IOPS' fields are required. For Azure, only the 'disktype' field is required. For Ultra disk type, you can provide all 3 fields: 'storage', 'IOPS' and 'auto-expansion'. For Premium type, you can set the 'auto-expansion' field only. For GCP, only the 'pd ssd' disk type is available, and you cannot set the 'IOPS' field. (See [below for nested schema](#nestedatt--service_groups--node--disk))
```

**Review by Rebecca-Martinez007 — CHANGES_REQUESTED** [review]
Made a few changes to the phrasing and wording. 

@simon-dew let me know if there any other changes you want to make.

**Review by simon-dew — DISMISSED** [review]
LGTM 👍

**Review by Rebecca-Martinez007 — APPROVED** [review]
LGTM!

**Review by SaicharanCB — APPROVED** [review]

**Review by Rebecca-Martinez007 — APPROVED** [review]
LGTM

**Review by PaulomeeCb — APPROVED** [review]

**Pull request merged**